### PR TITLE
move visit header and patient banner to patient chart

### PR DESCRIPTION
**Pull request by hadijahkyampeire**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [x] My work includes tests or is validated by existing tests.

## Summary
- Move the patient visit header to the patient chart.
- Add the patient banner to the patient chart as a component instead of an extension which makes it improve the performance

## Screenshots
Screenshot 2022-08-31 at 22 48 02

## Related Issue
[Investigate and reduce the loading time of patient banner](https://issues.openmrs.org/browse/O3-1251)

## Other
Part of removing it from its original place is in this PR. https://github.com/openmrs/openmrs-esm-patient-management/pull/340

### CODE REVIEW

File: packages/esm-patient-chart-app/src/banner/contact-details/contact-details.component.tsx
Comment (ZacButko) [inline packages/esm-patient-chart-app/src/banner/contact-details/contact-details.component.tsx:26]: TODO: These fields and their order should be loaded from config or Address Template XML
(Not your worry Hadijah)
New ticket -> https://issues.openmrs.org/browse/O3-1491

File: packages/esm-patient-chart-app/src/banner/patient-banner.component.tsx
Comment (ZacButko) [inline packages/esm-patient-chart-app/src/banner/patient-banner.component.tsx:102]: I don't think `lodash.capitalize` respects locales, so this might give a poor conversion for other languages.
TODO: Add capital gender to translation file.

File: packages/esm-patient-chart-app/src/banner/patient-banner.scss
Comment (ZacButko) [inline packages/esm-patient-chart-app/src/banner/patient-banner.scss:78]: This is dangerous. We need to clean all of these up
New Ticket -> https://issues.openmrs.org/browse/O3-1492

File: packages/esm-patient-chart-app/src/banner/ui-components/overflow-menu.component.tsx
Comment (ZacButko) [inline packages/esm-patient-chart-app/src/banner/ui-components/overflow-menu.component.tsx:47]: Using classnames from carbon is a good indication that we should probably be using their actual component. In this case https://react.carbondesignsystem.com/?path=/story/components-overflowmenu--default . @hadijahkyampeire are you able to clean this one? and for the child components as well. This seems like a lot of customization for what should be an out-of-the-box component from Carbon.

File: packages/esm-patient-chart-app/src/patient-chart/patient-chart.component.tsx
Comment (ZacButko) [inline packages/esm-patient-chart-app/src/patient-chart/patient-chart.component.tsx:48]: 💯 💯 💯  Awesome.

File: packages/esm-patient-chart-app/src/root.component.tsx
Comment (ZacButko) [inline packages/esm-patient-chart-app/src/root.component.tsx:31]: 👍

File: packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx
Comment (ZacButko) [inline packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx:28]: I think this can be changed to `patient.displayName` for better localization

File: packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx
Comment (ZacButko) [inline packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx:29]: Translate capital gender again

File: packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx
Comment (ZacButko) [inline packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx:83]: This is really weird syntax for this. This if statement should be brought outside the return and a fallback return should be provided.

File: packages/esm-patient-chart-app/src/visit-header/workspaces.ts
Comment (ZacButko) [inline packages/esm-patient-chart-app/src/visit-header/workspaces.ts:91]: I think I'm missing why this file is necessary.. Don't we already have `launchPatientWorkspaces` from [esm-patient-common-lib](https://github.com/openmrs/openmrs-esm-patient-chart/blob/main/packages/esm-patient-common-lib/src/workspaces/workspaces.ts) ? If this file is redundant let's remove it

**Review by ZacButko — CHANGES_REQUESTED** [review]
Great work, thank you for taking this on @hadijahkyampeire . Overall looks good and I know these are just porting files from esm-patient-management, but there are a few small things maybe we can clean up as we go.

The change I would want to see is to remove the duplicate `workspaces.ts` file if it is ok to use the one from from esm-patient-common-lib.

This PR is closely tied to https://github.com/openmrs/openmrs-esm-patient-management/pull/340. Let's wait until we have resolution on that one before we merge this

File: packages/esm-patient-chart-app/src/visit-header/workspaces.ts
Comment (hadijahkyampeire) [reply packages/esm-patient-chart-app/src/visit-header/workspaces.ts:91]: Yeah I have realized that, I will delete the workspace.ts and use the workspace from the common lib. Thanks for the catch.

File: packages/esm-patient-chart-app/src/banner/contact-details/contact-details.component.tsx
Comment (hadijahkyampeire) [reply packages/esm-patient-chart-app/src/banner/contact-details/contact-details.component.tsx:26]: I will follow up with the ticket above.

File: packages/esm-patient-chart-app/src/banner/patient-banner.scss
Comment (hadijahkyampeire) [reply packages/esm-patient-chart-app/src/banner/patient-banner.scss:78]: okay I will check it out too.

File: packages/esm-patient-chart-app/src/banner/patient-banner.component.tsx
Comment (hadijahkyampeire) [reply packages/esm-patient-chart-app/src/banner/patient-banner.component.tsx]: I have done the translation I don't know if I have done it right.

File: packages/esm-patient-chart-app/src/banner/contact-details/contact-details.component.tsx
Comment (ZacButko) [reply packages/esm-patient-chart-app/src/banner/contact-details/contact-details.component.tsx:26]: Oh! Sorry I just made a note of these so we do not lose track of them. You don't have to focus on them now

File: packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx
Comment (ZacButko) [reply packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx:28]: Hmm you're right. It looks like this isn't available anymore with fhir api? I can see it through REST endpoint for say visit - but even then its not always available
Screen Shot 2022-09-07 at 9 51 12 AM

File: packages/esm-patient-chart-app/src/banner/ui-components/overflow-menu.component.tsx
Comment (hadijahkyampeire) [reply packages/esm-patient-chart-app/src/banner/ui-components/overflow-menu.component.tsx:47]: This is somehow tricky because the carbon design component will not bring out the design as it is desired here on its own, we might need to combine it with custom `JSX` and that might require some more time to get it to work.

File: packages/esm-patient-chart-app/src/banner/contact-details/contact-details.component.tsx
Comment (vasharma05) [inline packages/esm-patient-chart-app/src/banner/contact-details/contact-details.component.tsx:47]: Please use a TextSkeleton for loading purposes. According to the designs, skeletons are to be shown when the data is loading.
Thanks!

File: packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx
Comment (vasharma05) [reply packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx:28]: Display is not available with the FHIR, its a property of the REST

**Review by vasharma05 — APPROVED** [review]

**Comment by vasharma05** [comment]
Great work @hadijahkyampeire!

File: packages/esm-patient-chart-app/src/banner/contact-details/contact-details.component.tsx
Comment (hadijahkyampeire) [reply packages/esm-patient-chart-app/src/banner/contact-details/contact-details.component.tsx:47]: @vasharma05 what is a TextSkeleton?

**Review by ZacButko — APPROVED** [review]

File: packages/esm-patient-chart-app/src/banner/patient-banner.component.tsx
Comment (hadijahkyampeire) [reply packages/esm-patient-chart-app/src/banner/patient-banner.component.tsx:13]: hmm, I think we need to revisit this banner sometime.

**Pull request merged**
